src/rard/research/tests/views/test_antiquarian.py

```diff
@@ -181,3 +182,58 @@ def test_ordered_queryset(self):
         qs = view.get_queryset()
         self.assertEqual(2, len(qs))
         self.assertEqual([a.pk for a in qs.all()], [a2.pk, a1.pk])
+
+
+class TestAntiquarianUpdateIntroductionView(TestCase):
+    def setUp(self):
+        self.antiquarian = Antiquarian.objects.create()
+        self.url = reverse("antiquarian:delete", kwargs={"pk": self.antiquarian.pk})
+        self.request = RequestFactory().get(self.url)
+        self.request.user = UserFactory.create()
+
+        self.antiquarian.lock(self.request.user)
+
+        self.response = AntiquarianUpdateIntroductionView.as_view()(
+            self.request, pk=self.antiquarian.pk
+        )
+
+    def test_context_data(self):
+        self.assertEqual(
+            self.response.context_data["text_object"], self.antiquarian.introduction
+        )
+        self.assertEqual(self.response.context_data["object_class"], "antiquarian")
+
+    def test_success_url(self):
+        # this is more complicated than with other views
+        # due to the conditional rendering on another view
+        # // I think
+        success_url = self.response.context_data["view"].get_success_url()
+        self.assertEqual(success_url, f"/antiquarian/{self.antiquarian.pk}/")
+
+    def test_update_intro(self):
+        """This checks that an introduction object is created
+        when the book object is created but is empty
+        and will update on POST"""
+        intro = self.response.context_data["object"].introduction
+
+        self.assertTrue(bool(intro.pk))
+        self.assertFalse(bool(intro.content))
+
+        ant_pk = self.response.context_data["object"].pk
+
+        intro_text = "testing update of introduction"
+
+        data = {
+            "introduction_text": intro_text,
+        }
+        post_request = RequestFactory().post(self.url, data=data)
+        post_request.user = self.request.user
+        AntiquarianUpdateIntroductionView.as_view()(post_request, pk=ant_pk)
+
+        view = AntiquarianUpdateIntroductionView.as_view()(
+            self.request, pk=self.antiquarian.pk
+        )
+
+        intro = view.context_data["object"].introduction
```

Comment:
Getting the updated intro this way seems a bit convoluted. Couldn't you just access self.antiquarian.introduction?

Reply:
it returns false if accessed directly like that but true through the view. Any ideas why that is?

Reply:
Sticking a `self.antiquarian.refresh_from_db()` in there should do it.

Reply:
🤦‍♀️